var/spack/repos/builtin/packages/singularityce/package.py

```diff
@@ -0,0 +1,188 @@
+        filter_file(r'^# mksquashfs path =',
+                    'mksquashfs path = {0}'.format(squash_path),
+                    join_path(prefix.etc, 'singularity', 'singularity.conf'))
+
+    #
+    # Assemble a script that fixes the ownership and permissions of several
+    # key files, install it, and tty.warn() the user.
+    # HEADSUP: https://github.com/spack/spack/pull/10412.
+    #
+    def perm_script(self):
+        return 'spack_perms_fix.sh'
+
+    def perm_script_tmpl(self):
+        return "{0}.j2".format(self.perm_script())
+
+    def perm_script_path(self):
+        return join_path(self.spec.prefix.bin, self.perm_script())
+
+    def _build_script(self, filename, variable_data):
+        with open(filename, 'w') as f:
+            env = spack.tengine.make_environment(dirs=self.package_dir)
+            t = env.get_template(self.perm_script_tmpl())
+            f.write(t.render(variable_data))
+
+    @run_after('install')
+    def build_perms_script(self):
+        if self.spec.satisfies('+suid'):
+            script = self.perm_script_path()
+            chown_files = ['libexec/singularity/bin/starter-suid',
+                           'etc/singularity/singularity.conf',
+                           'etc/singularity/capability.json',
+                           'etc/singularity/ecl.toml']
+            setuid_files = ['libexec/singularity/bin/starter-suid']
+            self._build_script(script, {'prefix': self.spec.prefix,
+                                        'chown_files': chown_files,
+                                        'setuid_files': setuid_files})
+            chmod = which('chmod')
+            chmod('555', script)
+
+    # Until tty output works better from build steps, this ends up in
+    # the build log.  See https://github.com/spack/spack/pull/10412.
+    @run_after('install')
+    def caveats(self):
+        if self.spec.satisfies('+suid'):
+            tty.warn("""
+            For full functionality, you'll need to chown and chmod some files
+            after installing the package.  This has security implications.
+            For details, see:
+            https://sylabs.io/guides/2.6/admin-guide/security.html
+            https://sylabs.io/guides/3.2/admin-guide/admin_quickstart.html#singularity-security
+
+            We've installed a script that will make the necessary changes;
+            read through it and then execute it as root (e.g. via sudo).
+
+            The script is named:
+
+            {0}
+            """.format(self.perm_script_path()))
+
+
+class Singularityce(SingularityBase):
+
+    homepage = "https://sylabs.io/singularity/"
+    url      = "https://github.com/sylabs/singularity/releases/download/v3.6.4/singularity-3.6.4.tar.gz"
+    git      = "https://github.com/sylabs/singularity.git"
+
+    maintainers = ['alalazo']
+    version('master', branch='master')
+
+    # SingularityBase covers up until 3.7.4
```

Comment:
```suggestion
    version('3.8.0', sha256='5fa2c0e7ef2b814d8aa170826b833f91e5031a85d85cd1292a234e6c55da1be1')
```

Reply:
oops should have looked ahead to save myself some work! I just added this version and updated url. Sorry about that.